plugins/notifications-backend/src/service/NotificationCleaner.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  LoggerService,
+  readSchedulerServiceTaskScheduleDefinitionFromConfig,
+  SchedulerService,
+  SchedulerServiceTaskScheduleDefinition,
+} from '@backstage/backend-plugin-api';
+import { Config, readDurationFromConfig } from '@backstage/config';
+import { NotificationsStore } from '../database';
+import { durationToMilliseconds, HumanDuration } from '@backstage/types';
+
+export class NotificationCleaner {
+  static readonly initNotificationCleaner = async (
```

Comment:
I don't know why but this static assignment in a class feels so odd :) Couldn't this just have been free functions or a normal class with a constructor

Reply:
This was to get rid of calling async function in constructor which is a bad habit. But replaced it with a new function to initialize the taskRunner.